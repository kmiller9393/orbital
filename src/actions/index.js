export const addGlossary = glossary => ({
  type: 'ADD_GLOSSARY',
  glossary
});

export const addTriviaSet = triviaSet => ({
  type: 'ADD_TRIVIA_SET',
  triviaSet
});

export const addScore = score => ({
  type: 'ADD_SCORE',
  score
});
