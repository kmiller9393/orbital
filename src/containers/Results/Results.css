.results-header {
  color: #f7f7f7;
  font-family: 'Chalkboard SE', 'Times New Roman', serif;
  font-weight: 100; }

.results-image {
  height: 55px;
  margin-top: 1.5rem;
  width: 125px; }

.score {
  color: #f7f7f7;
  font-size: 1.4rem; }

span {
  color: #00da59; }

.score-container {
  margin-top: 10rem; }
