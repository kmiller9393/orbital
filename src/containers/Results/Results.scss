.results-header {
  color: #f7f7f7;
  font-family: 'Chalkboard SE', 'Times New Roman', serif;
  font-weight: 100;
}

.results-image {
  height: 55px;
  margin-top: 1.5rem;
  width: 125px;
}

.score {
  color: #f7f7f7;
  font-size: 1.4rem;
}

.high-score {
  color: #00da59;
}

.low-score {
  color: #c90000;
}

.score-container {
  margin-top: 10rem;
}

.results-button {
  background-color: #009c92;
  border-color: #009c92;
  border-radius: 5px;
  box-shadow: 0px 8px 15px rgba(0, 0, 0, 0.5);
  color: #f7f7f7;
  cursor: pointer;
  font-family: 'Chalkboard SE', 'Times New Roman', serif;
  font-size: 1.4rem;
  height: 55px;
  margin-top: 2.5rem;
  text-decoration: none;
  text-shadow: 2px 4px 3px rgba(0, 0, 0, 0.3);
  width: 220px;
}

.results-button:active {
  background-color: #03aca1;
}
