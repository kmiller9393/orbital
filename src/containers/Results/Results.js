import React, { Component } from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { resetScore } from '../../actions';
import PropTypes from 'prop-types';
import orbital from '../../images/orbital-header.svg';
import './Results.css';

export class Results extends Component {
  constructor() {
    super();
    this.state = {};
  }
  render() {
    const { score, resetScore } = this.props;

    return (
      <div>
        <NavLink to="/" onClick={() => resetScore()}>
          <img className="results-image" src={orbital} alt="Orbital" />
        </NavLink>
        <div className="score-container">
          <h1 className="results-header">Your Results</h1>
          <p className="score">
            Score:{' '}
            <span className={score > 6 ? 'high-score' : 'low-score'}>
              {((score / 9) * 100).toFixed(2)}%
            </span>
          </p>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  triviaAnswers: state.triviaAnswers,
  score: state.score
});

export const mapDispatchToProps = dispatch => ({
  resetScore: () => dispatch(resetScore())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Results);

const { array, number, func } = PropTypes;
Results.propTypes = {
  triviaAnswers: array,
  score: number,
  resetScore: func
};
