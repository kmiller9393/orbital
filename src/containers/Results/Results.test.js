import React from 'react';
import { shallow } from 'enzyme';
import { Results, mapDispatchToProps, mapStateToProps } from './Results';
import { resetScore } from '../../actions';

describe('Results', () => {
  let wrapper;
  let mockResetScore;
  let mockScore;

  beforeEach(() => {
    mockResetScore = jest.fn();
    wrapper = shallow(
      <Results resetScore={mockResetScore} score={mockScore} />
    );
  });

  describe('Results component', () => {
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should call the resetScore action creator when the link is clicked', () => {
      wrapper.find('.reset-link').simulate('click');
      expect(mockResetScore).toHaveBeenCalled();
    });

    it('should have a class of high-score if the score is greater than 6', () => {
      wrapper = shallow(<Results score={7} />);
      expect(wrapper.find('span').hasClass('high-score')).toEqual(true);
      expect(wrapper.find('span').hasClass('low-score')).toEqual(false);
    });

    it('should have a class of low-score if the score is less than 6', () => {
      wrapper = shallow(<Results score={5} />);
      expect(wrapper.find('span').hasClass('low-score')).toEqual(true);
      expect(wrapper.find('span').hasClass('high-score')).toEqual(false);
    });
  });

  describe('mapStateToProps', () => {
    it('should return an array of triviaAnswer objects and the score', () => {
      const triviaAnswers = {
        question: 'a small solar system object composed mostly of rock',
        answer: 'Asteroid',
        id: 0
      };

      const mockState = {
        triviaAnswers,
        score: 0
      };
      const expected = {
        score: 0,
        triviaAnswers: {
          answer: 'Asteroid',
          id: 0,
          question: 'a small solar system object composed mostly of rock'
        }
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });

    describe('mapDispatchToProps', () => {
      it('should call dispatch when using a function from mapDispatchToProps', () => {
        const mockDispatch = jest.fn();
        const actionToDispatch = resetScore();

        const mappedProps = mapDispatchToProps(mockDispatch);
        mappedProps.resetScore();

        expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
      });
    });
  });
});
