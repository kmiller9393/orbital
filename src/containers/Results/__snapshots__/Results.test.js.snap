// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Results component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Results
    resetScore={[Function]}
    score={undefined}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="reset-link"
          onClick={[Function]}
          to="/"
>
          <img
                    alt="Orbital"
                    className="results-image"
                    src="orbital-header.svg"
          />
</NavLink>,
        <div
          className="score-container"
>
          <h1
                    className="results-header"
          >
                    Your Results
          </h1>
          <p
                    className="score"
          >
                    Score:
                     
                    <span
                              className="low-score"
                    >
                              NaN
                              %
                    </span>
          </p>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <img
            alt="Orbital"
            className="results-image"
            src="orbital-header.svg"
/>,
          "className": "reset-link",
          "onClick": [Function],
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Orbital",
            "className": "results-image",
            "src": "orbital-header.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="results-header"
>
              Your Results
</h1>,
            <p
              className="score"
>
              Score:
               
              <span
                            className="low-score"
              >
                            NaN
                            %
              </span>
</p>,
          ],
          "className": "score-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Your Results",
              "className": "results-header",
            },
            "ref": null,
            "rendered": "Your Results",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Score:",
                " ",
                <span
                  className="low-score"
>
                  NaN
                  %
</span>,
              ],
              "className": "score",
            },
            "ref": null,
            "rendered": Array [
              "Score:",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "NaN",
                    "%",
                  ],
                  "className": "low-score",
                },
                "ref": null,
                "rendered": Array [
                  "NaN",
                  "%",
                ],
                "type": "span",
              },
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="reset-link"
            onClick={[Function]}
            to="/"
>
            <img
                        alt="Orbital"
                        className="results-image"
                        src="orbital-header.svg"
            />
</NavLink>,
          <div
            className="score-container"
>
            <h1
                        className="results-header"
            >
                        Your Results
            </h1>
            <p
                        className="score"
            >
                        Score:
                         
                        <span
                                    className="low-score"
                        >
                                    NaN
                                    %
                        </span>
            </p>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <img
              alt="Orbital"
              className="results-image"
              src="orbital-header.svg"
/>,
            "className": "reset-link",
            "onClick": [Function],
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Orbital",
              "className": "results-image",
              "src": "orbital-header.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="results-header"
>
                Your Results
</h1>,
              <p
                className="score"
>
                Score:
                 
                <span
                                className="low-score"
                >
                                NaN
                                %
                </span>
</p>,
            ],
            "className": "score-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Your Results",
                "className": "results-header",
              },
              "ref": null,
              "rendered": "Your Results",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Score:",
                  " ",
                  <span
                    className="low-score"
>
                    NaN
                    %
</span>,
                ],
                "className": "score",
              },
              "ref": null,
              "rendered": Array [
                "Score:",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "NaN",
                      "%",
                    ],
                    "className": "low-score",
                  },
                  "ref": null,
                  "rendered": Array [
                    "NaN",
                    "%",
                  ],
                  "type": "span",
                },
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
